### DISCUSSION TITLE
Complete Guide for fine tunning ( from custrom dataset preparation to fine tuneing over that data  ) for Single speaker-no english speaker

### DISCUSSION THREAD

**Discussion by SanketDhuri** [yl4579/StyleTTS2#184] [General]
can some give complete and step by process to train this model as answers for following questions:

1. how to prepare custom dataset for fine tuning a single speaker ?
2. what are hyper parameters as best to ta finetune and inference no-English / Devnagari language like Marathi ?

**Comment by slooi** [comment]
Hey @SanketDhuri did you ever learn the steps to making a custom dataset? Currently I'm attempting to train my own voice model and was looking for a streamlined guide

**Reply by martinambrus** [reply]
depending on the dataset language needed, if it's English, you can try and check my preparation notebooks [here](https://github.com/yl4579/StyleTTS2/discussions/144)

**Reply by SanketDhuri** [reply]
@slooi I followed to the same guidelines provided [here](https://github.com/yl4579/StyleTTS2/discussions/144). I used audio editing tools to manually create custom data. I had to prepare the transcripts manually because I couldn’t find a suitable model that could simplify the transcription process. This was mainly because I was training a language that wasn’t supported by any of the available ASR models.

If you’re facing any hardware limitations, I recommend using Kaggle. I trained both stages on Kaggle.
